websocket-codec/src/mask.rs

```diff
@@ -49,75 +33,31 @@ fn mask_u8_in_place(data: &mut [u8], mut mask: u32) -> u32 {
     mask
 }
 
-fn mask_u8_copy(buf: &mut [u8], data: &[u8], mut mask: u32) -> u32 {
-    assert_eq!(buf.len(), data.len());
-
-    for (dest, &src) in buf.into_iter().zip(data) {
-        *dest = src ^ (mask as u8);
-        mask = mask.rotate_right(8);
-    }
-
-    mask
-}
-
 fn mask_aligned_in_place(data: &mut [u32], mask: u32) {
     for n in data {
         *n ^= mask;
     }
 }
 
-fn mask_aligned_copy(buf: &mut [u32], data: &[u32], mask: u32) {
-    assert_eq!(buf.len(), data.len());
-
-    for (dest, &src) in buf.into_iter().zip(data) {
-        *dest = src ^ mask;
-    }
-}
-
 /// Masks data sent by a client, and unmasks data received by a server.
-pub struct Masker {
```

Comment:
Note that a separate `Masker` type is now unnecessary as it no longer needs to hold `but: Bytes`. `Masker` can be reduced down to a single function within this module.

Reply:
yup checked out your implementation will modify